xrootd/xrdfs/stat.go

```diff
@@ -0,0 +1,180 @@
+	// kXR_posc is an option of open request indicating that the "Persist On Successful Close" processing is enabled and
+	// the file will be persisted only when it has been explicitly closed.
+	StatIsPOSCPending StatFlags = 64
+)
+
+// EntryStat holds the entry name and the entry stat information.
+type EntryStat struct {
+	EntryName   string    // EntryName is the name of entry.
+	HasStatInfo bool      // HasStatInfo indicates if the following stat information is valid.
+	ID          int64     // ID is the OS-dependent identifier assigned to this entry.
+	EntrySize   int64     // EntrySize is the decimal size of the entry.
+	Flags       StatFlags // Flags identifies the entry's attributes.
+	Mtime       int64     // Mtime is the last modification time in Unix time units.
+}
+
+// Name implements os.FileInfo.
+func (es EntryStat) Name() string {
+	return es.EntryName
+}
+
+// Size implements os.FileInfo.
+func (es EntryStat) Size() int64 {
+	return es.EntrySize
+}
+
+// ModTime implements os.FileInfo.
+func (es EntryStat) ModTime() time.Time {
+	return time.Unix(es.Mtime, 0)
+}
+
+// Sys implements os.FileInfo.
+func (es EntryStat) Sys() interface{} {
+	return nil
+}
+
+// Mode implements os.FileInfo.
+func (es EntryStat) Mode() os.FileMode {
+	var mode os.FileMode
+	if es.IsDir() {
+		mode |= os.ModeDir
+	}
+	if es.IsWritable() {
+		mode |= 0222
+	}
+	if es.IsReadable() {
+		mode |= 0444
+	}
+	return mode
+}
+
+// IsExecutable indicates whether this entry is either an executable file or a searchable directory.
+func (es EntryStat) IsExecutable() bool {
+	return es.Flags&StatIsExecutable != 0
+}
+
+// IsDir indicates whether this entry is a directory.
+func (es EntryStat) IsDir() bool {
+	return es.Flags&StatIsDir != 0
+}
+
+// IsOther indicates whether this entry is neither a file nor a directory.
+func (es EntryStat) IsOther() bool {
+	return es.Flags&StatIsOther != 0
+}
+
+// IsOffline indicates whether this the file is not online (i. e., on disk).
+func (es EntryStat) IsOffline() bool {
+	return es.Flags&StatIsOffline != 0
+}
+
+// IsReadable indicates whether this read access to that entry is allowed.
+func (es EntryStat) IsReadable() bool {
+	return es.Flags&StatIsReadable != 0
+}
+
+// IsWritable indicates whether this write access to that entry is allowed.
+func (es EntryStat) IsWritable() bool {
+	return es.Flags&StatIsWritable != 0
+}
+
+// IsPOSCPending indicates whether this the file was created with kXR_posc and has not yet been successfully closed.
+// kXR_posc is an option of open request indicating that the "Persist On Successful Close" processing is enabled and
+// the file will be persisted only when it has been explicitly closed.
+func (es EntryStat) IsPOSCPending() bool {
+	return es.Flags&StatIsPOSCPending != 0
+}
+
+// MarshalXrd implements xrdproto.Marshaler.
+func (o EntryStat) MarshalXrd(wBuffer *xrdenc.WBuffer) error {
+	if !o.HasStatInfo {
```

Comment:
shouldn't `EntryName` be written out?

Reply:
It should be written out in the `dirlist` response, yes. However, in `open` and `stat` responses entry name isn't specified. So, I thought that we can leave that to the specific response Marshaler.